### DISCUSSION TITLE
can we change style of HTML video <track> tag?

### DISCUSSION THREAD

**Discussion by MonaAghili** [tailwindlabs/tailwindcss#14272] [Help] [answered]
Is there any way to apply color and font classes to the `<track>` tag using Tailwind to style the captions of a video? The only way to change the style via CSS is by using the `::cue` pseudo-element. But I don't know if Tailwind provides a solution for this.

```css
::cue {
  color: #ccc;
}
```

**Comment by wongjn** [comment]
You can use an [arbitrary variant](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) like:

```html
<video class="[&::cue]:text-neutral-200">
```
Or [add it as a variant in a custom Tailwind plugin](https://tailwindcss.com/docs/plugins#adding-variants).

**Reply by MonaAghili** [reply]
Thanks.
The text size class `<video class="[&::cue]:text-xl">` works fine, but the classes `[&::cue]:bg-white` and `[&::cue]:text-[#1E1E1E]` aren't applying to the subtitle as expected.

**Reply by wongjn** [reply, marked as answer]
Using [this](https://play.tailwindcss.com/ERTNjUQ7ET) as a test:
- In Firefox, seems to work fine:  
  ![image](https://github.com/user-attachments/assets/8bc4e326-7a2d-4df9-b1ac-111652efd0f8)
- In Chrome, indeed seems broken:
  ![image](https://github.com/user-attachments/assets/7f1ce308-3a0e-4f57-8e07-8388e083f00c)

After some experimenting, it seems to be the CSS variable usage tripping up Chrome. As a work-around, you could consider:
- Using [arbitrary property classes](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties): https://play.tailwindcss.com/Lcid5bU9dM
- Disabling `backgroundOpacity` and `textOpacity` core plugins to remove the CSS variable usage: https://play.tailwindcss.com/znOz48QoUw?file=config

**Reply by MonaAghili** [reply]
Oh thank you.

This approach seems to change the background, though it's not perfect, but it works.
https://play.tailwindcss.com/TdkHVY6O69
